client.go

```diff
@@ -64,25 +66,33 @@ func buildAzureClient() (*AzureClient, error) {
 		return nil, fmt.Errorf("Error building ARM Client: %s", err)
 	}
 
+	api, err := environments.EnvironmentFromString("")
+	if err != nil {
+		return nil, fmt.Errorf("unable to find environment %q from endpoint %q: %+v", environmentName, endpoint, err)
+	}
+
 	sender := sender.BuildSender("Azure Dalek")
 
 	oauthConfig, err := client.BuildOAuthConfig(environment.ActiveDirectoryEndpoint)
 	if err != nil {
 		return nil, err
 	}
 
-	resourceManagerAuth, err := client.GetAuthorizationToken(sender, oauthConfig, environment.TokenAudience)
+	resourceManagerAuth, err := client.GetMSALToken(context.Background(), api.ResourceManager, sender, oauthConfig, environment.TokenAudience)
```

Comment:
we probably want to use a time'd context here:

```suggestion
	ctx, cancel := context.WithDeadline(context.Background(), 30 * time.Second())
	defer cancel()
	resourceManagerAuth, err := client.GetMSALToken(ctx, api.ResourceManager, sender, oauthConfig, environment.TokenAudience)
```